base/base/isSharedPtrUnique.h

```diff
@@ -0,0 +1,9 @@
+#pragma once
+
+#include <memory>
+
+template <typename T>
+bool isSharedPtrUnique(const std::shared_ptr<T> & ptr)
```

Comment:
`std::shared_ptr<T>::unique()` is removed in C++20 https://en.cppreference.com/w/cpp/memory/shared_ptr/unique